prometheus/config.go

```diff
@@ -148,7 +153,20 @@ func (c Configuration) NewReporter(
 		mux := http.NewServeMux()
 		mux.Handle(path, reporter.HTTPHandler())
 		go func() {
-			if err := http.ListenAndServe(addr, mux); err != nil {
+			nwk := c.ListenNetwork
+			if nwk == "" {
+				nwk = "tcp"
+			}
+
+			lsn, err := net.Listen(nwk, addr)
+			if err != nil {
+				opts.OnRegisterError(err)
+				return
+			}
```

Comment:
can you keep the long-form names (`network`, `listener`)

Reply:
@vdarulis thank you. updated.